jiant/tasks/tasks.py

```diff
@@ -3731,3 +3732,128 @@ def get_metrics(self, reset=False):
         """Get metrics specific to the task"""
         acc = self.scorer1.get_metric(reset)
         return {"accuracy": acc}
+
+
+@register_task("wikipedia_corpus_sop", rel_path="wikipedia_corpus_small")
+class SentenceOrderTask(Task):
+    """ Task class for Sentence Order Prediction (SOP) with Wikipedia data (more specifically,
+        the dump from Wikimedia). See the ALBERT paper for details on SOP: https://arxiv.org/abs/1909.11942.
+        We are currently using an unpreprocessed version of the Wikipedia corpus
+        that consists of 5% of the data. You can generate the data by following the
+        instructions from jiant/scripts/mlm.
+    """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        super(SentenceOrderTask, self).__init__(name, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+        self.n_classes = 2
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+        self.files_by_split = {
+            "train": os.path.join(path, "train.txt"),
+            "val": os.path.join(path, "valid.txt"),
+            "test": os.path.join(path, "test.txt"),
+        }
+        self.val_metric_decreases = False
+        self.val_metric = "%s_accuracy" % self.name
+        self.acc_scorer = BooleanAccuracy()
+
+    def get_metrics(self, reset=False):
+        """Get metrics specific to the task"""
+        acc = self.acc_scorer.get_metric(reset)
+        return {"accuracy": acc}
+
+    def update_metrics(self, out, batch):
+        logits = out["logits"]
+        labels = out["labels"]
+        _, preds = logits.max(dim=1)
+        self.acc_scorer(preds, labels)
+
+    def get_data_iter(self, path):
+        """Loading data file and tokenizing the text
+        Args:
+            path: (str) data file path
+        """
+        import csv
+
+        f = open(path, "r")
+        reader = csv.reader(f)
+        text = list(reader)
```

Comment:
This looks like a list of file lines, not sentences. Clarify where this gets segmented into sentences.

Reply:
This is segmented in the source text.

Reply:
Make it clear how to prepare the source text, then. Also, make sure it's a separate file from the MLM source—MLM requires full paragraphs, while SOP requires sentences.

Reply:
We actually use the same sentence level data for MLM. We've tried with both sentence level and paragraph level for MLM and we didn't see noticable difference in performance.